packages/ast/test/transform/denormalize-type-references/index.js

```diff
@@ -0,0 +1,51 @@
+// @flow
+const {
+  transform
+} = require("../../../lib/transform/denormalize-type-references");
+const { readFileSync } = require("fs");
+const path = require("path");
+const diff = require("jest-diff");
+const { NO_DIFF_MESSAGE } = require("jest-diff/build/constants");
+
+const compareMultilineString = (a, b) => {
+  a = JSON.stringify(a, null, 2);
+  b = JSON.stringify(b, null, 2);
+  const out = diff(a.trim(), b.trim());
+  if (out !== null && out !== NO_DIFF_MESSAGE) {
+    throw new Error("\n" + out);
+  }
+  if (a.trim() !== b.trim()) {
+    throw new Error("Assertion error");
+  }
+};
+
+const loadJSON = filename =>
+  JSON.parse(readFileSync(path.join(__dirname, filename), "utf8"));
+
+const copy = json => JSON.parse(JSON.stringify(json));
```

Comment:
I think that we can use `const copy = o => Object.assign({}, o);` here.

Reply:
I think that will just perform a shallow copy?

Reply:
That's right. I was a bit worried about perf but it's in the test so doesn't matter so much :+1: